ideas/new-ops.txt

```diff
@@ -0,0 +1,117 @@
+# Proposal for new stdlib operations
+
+This is a proposal for several APIs that could be added to stdlib.
+
+## `builtin` candidates
+
+Entity	Signature	Notes
+
+`Not`	`Not(x bool) (x bool)`	Return logical not operation applied to input `x`
+`Pair<A, B>`	`struct { first A; second B }`	A generic pair of items. Can stand in for other pair types we've introduced such as `ZipResult`
+`OrDefault<T>`	`OrDefault<T>(t T)`	Try to return `t` or `default` if `t` is unavailable. `default` can be injected with `IInjectable<T>`
+`Unzip<T, R>`	`Unzip<T, R>(seq stream<ZipResult<T, R>>) (first stream<T>, second stream<R>)`	The inverse of `Zip<T, R>` which is already in `builtin`
+
+## DI Interfaces
+
+Inject these no-inport interfaces rather than having it in the signature.
+
+If lambda components are ever supported, we can write this more nicely.
+
+```
+// Before
+rand.Bool
+// ...
+0.3 -> bool:p
+bool -> :out
+
+// After
+rand.Bool{() -> 0.3}
```

Comment:
I'm not sure if I get this one. Why are use mixing integers and booleans? I know it's just an example but would be nice to understand it.

Reply:
`p` is the probability of true in this case. `(1-p)` Of false